File: threedollar-api-user/src/main/java/com/depromeet/threedollar/api/user/config/security/WebSecurityConfig.java
package com.depromeet.threedollar.api.user.config.security;

import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
import org.springframework.security.config.http.SessionCreationPolicy;

@EnableWebSecurity
@Configuration
public class WebSecurityConfig extends WebSecurityConfigurerAdapter {

    @Override
    protected void configure(HttpSecurity http) throws Exception {
        http.authorizeRequests()
            .antMatchers("/monitoring/**").authenticated()
                .anyRequest().permitAll()
            .and()
                .requestMatchers((matcher) -> matcher.antMatchers("/h2-console/**")).headers().frameOptions().sameOrigin()
            .and()
                .httpBasic()
            .and()
                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
            .and()
                .csrf().disable();
    }

}

File: threedollar-api-admin/src/main/kotlin/com/depromeet/threedollar/api/admin/config/security/WebSecurityConfig.kt
package com.depromeet.threedollar.api.admin.config.security

import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
import org.springframework.security.config.http.SessionCreationPolicy

@EnableWebSecurity
@Configuration
class WebSecurityConfig : WebSecurityConfigurerAdapter() {

    override fun configure(http: HttpSecurity) {
        http.authorizeRequests()
                .antMatchers("/monitoring/**").authenticated()
                    .anyRequest().permitAll()
            .and()
                .requestMatchers {   it.antMatchers(   "/h2-console/**") }.headers().frameOptions().sameOrigin()
            .and()
                .httpBasic()
            .and()
                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
            .and()
                .csrf().disable()
    }

}
